pulpcore/cli/common.py

```diff
@@ -76,16 +75,53 @@ def wait_for_task(self, task_href: str, timeout: int = 120) -> Any:
             task = self.api.call("tasks_read", parameters={"task_href": task_href})
             if task["state"] == "completed":
                 return task
-            if task["state"] == "failed":
+            elif task["state"] == "failed":
                 raise click.ClickException(
                     f"Task {task_href} failed: '{task['error']['description']}'"
                 )
-            if task["state"] == "canceled":
+            elif task["state"] == "canceled":
                 raise click.ClickException("Task canceled")
-            click.echo(".", nl=False, err=True)
+            elif task["state"] in ["waiting", "running"]:
+                click.echo(".", nl=False, err=True)
+            else:
+                raise NotImplementedError(f"Unknown task state: {task['state']}")
         raise click.ClickException("Task timed out")
 
 
+class PulpEntityContext:
+    HREF: str = "entity"
+    LIST_ID: str = "tasks_list"
+    READ_ID: str = "tasks_read"
```

Comment:
Did you mean to hardcode these?

Reply:
No, but i needed to provide something.

Reply:
Should have been "entities_list" which will produce an error. I would have liked to make it an abstract class.